cloud/aws/actuators/machine/actuator.go

```diff
@@ -30,7 +32,7 @@ import (
 // ec2Svc are the functions from the ec2 service, not the client, this actuator needs.
 // This should never need to import the ec2 sdk.
 type ec2Svc interface {
-	CreateInstance(*clusterv1.Machine) (*ec2svc.Instance, error)
+	CreateInstance(*clusterv1.Machine, *v1alpha1.AWSMachineProviderConfig, *v1alpha1.AWSClusterProviderStatus) (*ec2svc.Instance, error)
```

Comment:
I've been struggling with this on my cloud-init integration work as well. In addition to info that is currently stored in the cluster providerstatus, we also need some of the info that is available within the base cluster object as well. I think it would make sense to unify the machine and cluster info needed into a consolidated struct rather than continuing to add additional disparate structs.

Reply:
decision was to defer to after MVP work